openhgnn/trainerflow/MeiRec_trainer.py

```diff
@@ -0,0 +1,206 @@
+from collections import OrderedDict
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+from tqdm import tqdm
+from ..models import build_model
+from . import BaseFlow, register_flow
+from ..tasks import build_task
+
+
+def cycle(iterable):
+    while True:
+        for x in iterable:
+            yield x
+
+
+@register_flow("MeiREC_trainer")
+class MeiRECTrainer(BaseFlow):
+
+    def __init__(self, args):
+        super().__init__(args)
+        self.args = args
+        self.model_name = args.model
+        self.device = args.device
+        self.task = build_task(args)
+
+        self.model = build_model(self.model_name).build_model_from_args(
+            self.args).model
+        print("build_model_finish")
+        self.model = self.model.to(self.device)
+        self.loss_fn = nn.BCELoss(reduction='mean')
+        self.evaluator = self.task.evaluate
+        self.optimizer = optim.Adam(self.model.parameters(), lr=args.lr)
+        self.scheduler = optim.lr_scheduler.ExponentialLR(self.optimizer, 0.99)
+        self.patience = args.patience
+        self.max_epoch = args.max_epoch
+        self.gpu_support = True if torch.cuda.is_available() else False
```

Comment:
I think we have already specified the GPU information, so this one is of no use, correct this and the following use.